infra/modules/providers/azure/app-service/main.tf

```diff
@@ -43,23 +55,23 @@ resource "azurerm_app_service" "appsvc" {
 }
 
 resource "null_resource" "acr_webhook_creation" {
-  count      = var.docker_enable_ci == true && var.azure_container_registry_name != "" ? length(keys(var.app_service_name)) : 0
+  count      = var.docker_enable_ci == true && var.azure_container_registry_name != "" ? length(keys(var.app_service_config)) : 0
   depends_on = [azurerm_app_service.appsvc]
 
   triggers = {
-    images_to_deploy = "${join(",", values(var.app_service_name))}"
+    images_to_deploy = "${join(",", [for k, v in var.app_service_config : v.image])}"
```

Comment:
Really cool! haven't seen this before

Reply:
With fresh eyes, i'm actually surprised this worked. I'm definitely going to have to look at this again. Thanks for noticing though.

Reply:
Due to splitting up the map, this is now an array that holds configurations without the keys.

```hcl
images_to_deploy = "${join(",", [for config in local.app_configs : config.image])}"
```